### PR TITLE
bulk add tags to keep

### PR DESCRIPTION
**Pull request by eishay**
@andrewconner @joshm1 @ahsu1230

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala [hunk 1]
```diff
@@ -651,6 +651,16 @@ class KeepsCommander @Inject() (
     addToCollection(tag.id.get, bookmarks) // why doesn't this update search?
   }
 
+  def tagKeeps(tag: Collection, userId: Id[User], keepIds: Seq[ExternalId[Keep]])(implicit context: HeimdalContext): (Seq[Keep], Seq[Keep]) = {
+    val (canEditKeep, cantEditKeeps) = db.readOnlyMaster { implicit s =>
+      keepIds map keepRepo.get partition { keep =>
+        keep.libraryId.exists(libraryCommander.canModifyLibrary(_, userId))
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala:657]: Expensive. Should do a `groupBy(_.libraryId)` first.

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala:657]: sure

**Comment by ahsu1230** [comment]
:+1: thanks Eishay. I think we have multiple tag multiple keeps (1 keep -> multiple tags) endpoints...
/site/collections/:id/addKeeps
/site/keeps/tag

both in KeepsController & both take in collectionId (one in URL, other in POSt body) so we should probably remove some endpoints that overlap in the future =)

**Comment by eishay** [comment]
can you please create an asana task for those for post launch?

**Pull request merged**